### PR TITLE
Setup turborepo

### PR DESCRIPTION
**Pull request by cotemper**
### Description

- Upgrade to yarn v3.2.0
- Use yarn workspace settings
- Use turborepo

### CODE REVIEW

File: rust/package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
-  "name": "@metaplex/metaplex-programs",
+  "name": "@raindrops-protocol/raindrops-programs",
   "version": "1.0.0",
   "private": true,
-  "description": "Metaplex Program Tests",
+  "description": "Raindrops Programs",
   "license": "MIT",
```
Comment (long-banana) [inline rust/package.json:6]: AGPL 3.0

File: rust/package.json [hunk 2]
```diff
@@ -17,8 +17,11 @@
     "typescript": "^4.3.5"
   },
   "scripts": {
-    "idl": "node test/idlToTs",
-    "test": "env MY_WALLET=$HOME/.config/solana/id.json ts-mocha -p ./tsconfig.json -t 1000000 test/*.ts"
+    "build": "anchor build --skip-lint",
```
Comment (long-banana) [inline rust/package.json:20]: Let's not do `--skip-lint`. We need to actually fix those at some point, and this is going to hide the pain.

File: turbo.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "lint": {
+      "outputs": []
+    },
+
+    "lint:check": {
+      "outputs": []
+    },
+    "deploy": {
+      "dependsOn": ["build", "test", "lint"],
```
Comment (long-banana) [inline turbo.json:22]: Does the order matter? Can we move lint forward in the list to fail a build if linting doesn't pass before running all the tests?

**Review by long-banana — CHANGES_REQUESTED** [review]
Couple of small changes. I'm not qualified to review the yarn workspaces parts. I've never used it. @rufflian, can you take a look and be the approver on those parts? Given our previous chats, you seem more qualified then me in that area.

File: js/cli/package.json [hunk 4]
```diff
@@ -36,7 +37,7 @@
     "@project-serum/anchor": "^0.24.2",
     "@raindrop-studios/sol-command": "^0.4.0",
     "@raindrop-studios/sol-kit": "^0.1.78",
-    "@raindrops-protocol/raindrops": "^0.0.7",
+    "@raindrops-protocol/raindrops": "workspace:js/lib",
```
Comment (rufflian) [inline js/cli/package.json:40]: I took a quick peek at the docs for workspace resolution and yarn is still listing this as "experimental syntax" - recommended is to use `workspace:*`
https://yarnpkg.com/features/workspaces#workspace-ranges-workspace

Tested the lib locally with yarn link and it seems to work well. Super quick to build with yarn 3 too!

File: turbo.json (hunk 3 above)
Comment (cotemper) [reply turbo.json:22]: Order doesn't matter. Deploy will run after all 'build', 'test', 'lint'  have pass.
Deploy can be publishing client library to npm, deploy program to devnet, etc.

Comment (long-banana) [reply turbo.json:22]: We can't enforce that linting is successful before running the tests?

Comment (cotemper) [reply turbo.json:22]: Yeah, we can. I will update the pipeline.
The order should be lint -> build -> test, right?

Comment (long-banana) [reply turbo.json:22]: Yup

Comment (cotemper) [reply turbo.json]: Updated

**Review by long-banana — COMMENTED** [review]
This looks good. May our builds be 🏃‍♂️💨🔥

**Pull request merged**